src/main/groovy/net/minecraftforge/gradleutils/GenerateActionsWorkflow.groovy

```diff
@@ -0,0 +1,128 @@
+import org.gradle.api.DefaultTask
+import org.gradle.api.Project
+import org.gradle.api.file.RegularFileProperty
+import org.gradle.api.provider.ListProperty
+import org.gradle.api.provider.Property
+import org.gradle.api.provider.ProviderFactory
+import org.gradle.api.tasks.Input
+import org.gradle.api.tasks.Optional
+import org.gradle.api.tasks.OutputFile
+import org.gradle.api.tasks.TaskAction
+import org.gradle.api.tasks.TaskProvider
+import org.yaml.snakeyaml.DumperOptions
+import org.yaml.snakeyaml.Yaml
+
+import javax.inject.Inject
+
+@CompileStatic
+abstract class GenerateActionsWorkflow extends DefaultTask {
+    private static final String NAME = 'generateActionsWorkflow'
+
+    static TaskProvider<GenerateActionsWorkflow> register(Project project) {
+        register(project, NAME)
+    }
+
+    static TaskProvider<GenerateActionsWorkflow> register(Project project, String name) {
+        project.tasks.register(name, GenerateActionsWorkflow)
+    }
+
+    @Inject
+    abstract ProviderFactory getProviders()
+
+    GenerateActionsWorkflow() {
+        this.description = 'Generates the GitHub Actions workflow file for the project, respecting declared subprojects in Git Version.'
+
+        this.outputFile.convention this.project.rootProject.layout.projectDirectory.file("build_${this.project.name}.yaml")
+
+        this.branch.convention this.providers.provider { this.project.extensions.getByType(GitVersionExtension).version.info.branch }
+        this.localPath.convention this.providers.provider { this.project.extensions.getByType(GitVersionExtension).version.projectPath }
+        this.paths.convention this.providers.provider { this.project.extensions.getByType(GitVersionExtension).version.subprojectPaths.collect { "!${it}/**".toString() } }
+        this.tagPrefix.convention this.providers.provider { this.project.extensions.getByType(GitVersionExtension).version.tagPrefix }
+        this.gradleJavaVersion.convention 21
+        this.sharedActionsBranch.convention 'v0'
+    }
+
+    @OutputFile
+    abstract RegularFileProperty getOutputFile()
+
+    @Input
+    @Optional
+    abstract Property<String> getBranch()
+
+    @Input
+    @Optional
+    abstract Property<String> getLocalPath()
+
+    @Input
+    @Optional
+    abstract ListProperty<String> getPaths()
+
+    @Input
+    @Optional
+    abstract Property<String> getTagPrefix()
+
+    @Input
+    abstract Property<Integer> getGradleJavaVersion()
+
+    @Input
+    abstract Property<String> getSharedActionsBranch()
+
+    @TaskAction
+    void exec() throws IOException {
+        var push = ['branches': this.branch.getOrElse('master')] as Map<String, ?>
+
+        var localPath = this.localPath.orNull
+        var paths = this.paths.getOrElse(Collections.emptyList())
+        var tagPrefix = this.tagPrefix.orNull
+
+        push.put('paths', new ArrayList<String>().tap {
+            if (localPath) add(localPath + '/**')
+
+            add('!settings.gradle')
```

Comment:
Should filter out !workflows as well probably

Reply:
I believe workflows are already filtered, but I'll add it if I'm wrong.